docs: Update oep-0042-bp-authentication

moeez96 opened the pull request.

robrap reviewed (commented):
Looks good. Thanks. OEPs Should now have a reverse order Change History section. Please add this change, and if the section doesn’t exist, you could add a note about the initial OEP.

moeez96 commented:
Thanks @robrap for the review. I did not quite understand about adding a note about the initial OEP, so I added the Change history section and added a note for the current change.

robrap commented on oeps/best-practices/oep-0042-bp-authentication.rst:
```diff
@@ -286,3 +283,9 @@ References
+Change History
+**************
+
+2022-11-23: Update for use of JWT in mobile applications
```
Here's a possible update:
```suggestion
2022-11-23
==========

* Update for use of JWT in mobile applications

2021-08-23
==========

* Introduced a link to mobile auth docs.

2020-05-26
==========

* Improved summary and minor text updates.

2020-05-15
==========

* Initial version of OEP was accepted.
```
